src/raster/gcp.rs

```diff
@@ -1,8 +1,111 @@
 //! Raster ground control point support
 
+use std::ffi::{CStr, CString};
+use std::marker::PhantomData;
+
+use gdal_sys::CPLErr;
+
+use crate::errors::Result;
 use crate::spatial_ref::SpatialRef;
+use crate::utils::{_last_cpl_err, _string};
 use crate::Dataset;
 
+/// An owned Ground Control Point.
+#[derive(Clone, Debug, PartialEq, PartialOrd)]
+pub struct Gcp {
+    /// Unique identifier, often numeric.
+    pub id: String,
+    /// Informational message or empty string.
+    pub info: String,
+    /// Pixel (x) location of GCP on raster.
+    pub pixel: f64,
+    /// Line (y) location of GCP on raster.
+    pub line: f64,
+    /// X position of GCP in georeferenced space.
+    pub x: f64,
+    /// Y position of GCP in georeferenced space.
+    pub y: f64,
+    /// Elevation of GCP, or zero if not known.
+    pub z: f64,
+}
+
+/// A wrapper over a Ground Control Point, borrowed from an existing dataset.
+#[derive(Clone, Debug)]
+#[repr(transparent)]
+pub struct GcpRef<'a> {
+    inner: gdal_sys::GDAL_GCP,
+    _phantom: PhantomData<&'a gdal_sys::GDAL_GCP>,
+}
```

Comment:
While this is called a `GcpRef`, in reality this looks more like a "newtype plus lifetime", which feels more "owned" than a "ref". So why is `inner` a `GDAL_GCP`, rather than a `* const GDAL_GCP`?

Also, why does the `repr` matter here given this is claiming to be a "ref"?

Reply:
Gotcha. I think I"m reading too much in the name, and have gotten myself embedded too deeply in the `foreign_types` semantics, where the borrowed `...Ref` type wraps an opaque pointer, and the owned type wraps (or can wrap) the actual structure.

Reply:
Maybe `BorrowedGcp` or `NonOwnedGcp` would be clearer?